src/util/fs.js

```diff
@@ -360,6 +360,11 @@ async function buildActionsForHardlink(
     const {src, dest} = data;
     const onFresh = data.onFresh || noop;
     const onDone = data.onDone || noop;
+    if (files.has(dest)) {
```

Comment:
What happens if an hardlinked file needs to change? Will it be removed in an earlier pass? Unless it is, maybe we could try comparing the inodes as well